rasa/engine/graph.py

```diff
@@ -436,10 +440,27 @@ def __call__(
         Returns:
             The node name and its output.
         """
-        received_inputs: Dict[Text, Any] = dict(inputs_from_previous_nodes)
+        # filter out arguments that dask couldn't lookup
+        received_inputs: Dict[Text, Any] = {}
+        for i in inputs_from_previous_nodes:
+            if isinstance(i, tuple):
+                arg_name, arg_value = i
+                received_inputs[arg_name] = arg_value
+            else:
+                logger.warning(
+                    f"Received unresolved argument: '{i}'. "
+                    f"Another component should have provided this as an "
+                    f"output."
+                )
```

Comment:
Something seems off about this section 🤔. At least the naming should be improved.

1. `inputs_form_previous_nodes` is a mapping of node name to output of a node according to the doc string of `__call__`.
2. in line 447, an element of this mapping is deconstructed into `arg_name` and `arg_value`. Judging from the previous observation this should be probably `node_name` and `node_output`
3. In line 451 a warning is issued using the `Text` type inputs from previous node as the name of an argument when the other elements and the doc string hint at the items in this list to be `node_name` and `node_output` pairs.

Reply:
sounds good, renamed the args. yes the description isn't entirely correct. as you can see I changed the type annotation as for node names that couldn't be resolved there is no tuple but just the node name.